andela/Ah-backend-aquaman
# #164069237 users should be able to bookmark articles for reading later

## What does this PR do?

This PR adds a feature that enables a user to bookmark an article for reading later, view a list of their bookmarks and delete a bookmarked article from their list.

## Description of the task to be completed

- Enable a user to bookmark an article
- Enable a user to view their bookmark list
- Enable a user to unbookmark an article

## How should this be manually tested?

- Clone this repository
- cd into the directory Ah-backend-aquaman
- Checkout the branch ft-bookmark-article-164069237
- Create and activate your virtual environment
- pip install -r requirements.txt
- python manage.py makemmigrations
- python manage.py migrate
- Source your .env file
- python manage.py runserver
- Sign up, verify your email and login
- To bookmark an article navigate to the endpoint http://127.0.0.1:8000/api/articles/articleslug/bookmark/ and make a post request
- To retrieve a list of your bookmarks go to http://127.0.0.1:8000/api/bookmarks/ and make a get request
- To delete a bookmarked article from your bookmarks list http://127.0.0.1:8000/api/articles/articleslug/bookmark/ and make a delete request

## What are the relevant pivotal tracker stories?

[#164069237](https://www.pivotaltracker.com/story/show/164069237)

## Screenshots

Ensure to paste your auth token into the token field for all the following endpoints

To bookmark an article, copy the article slug and attach it to your URL as shown

![image](https://user-images.githubusercontent.com/42435346/54631216-95c36c00-4a8c-11e9-808a-7240777ddc9c.png)

To get a list of your bookmarks

![image](https://user-images.githubusercontent.com/42435346/54632790-eb4d4800-4a8f-11e9-844e-a3259ca58b75.png)

To unbookmark an article

![image](https://user-images.githubusercontent.com/42435346/54631289-c0adc000-4a8c-11e9-935d-ed0b3c1ed58d.png)
